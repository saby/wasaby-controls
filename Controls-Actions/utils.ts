import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    document?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
