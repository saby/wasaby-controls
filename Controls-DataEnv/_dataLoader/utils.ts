import { loadAsync } from 'WasabyLoader/ModulesLoader';
import { addPageDeps } from 'UI/Deps';
import { logger as Logger } from 'Application/Env';
import { RPC } from 'Browser/Transport';

function resolveLogger(error: Error): typeof Logger.error | typeof Logger.warn {
    let logger = Logger.error;

    if (error instanceof RPC.Error) {
        logger = error.errType === 'warning' ? Logger.warn : Logger.error;
    }

    return logger;
}

export async function loadModule<TModule>(moduleName: string): Promise<TModule | undefined> {
    try {
        const module = await loadAsync<TModule>(moduleName);
        addPageDeps([moduleName]);
        return module;
    } catch (error: unknown) {
        loadErrorHandler(
            error,
            `Controls-DataEnv/dataLoader::Ошибка при загрузке модуля ${moduleName}`
        );
    }
}

export function loadErrorHandler(e: unknown, message: string): void {
    if (e instanceof Error) {
        const logger = resolveLogger(e);
        logger(`${message} ${e.message}`);
    }
}
