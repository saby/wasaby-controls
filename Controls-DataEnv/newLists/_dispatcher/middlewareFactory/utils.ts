export const addName = <T extends Function>(label: string, fn: T): T => {
    const wrapper = {
        [label]: ((...args: unknown[]) => fn(...args)) as unknown as T,
    };

    return wrapper[label];
};

const POSTFIX = '{name}';
export const createMask = <T extends string>(prefix: T) => `${prefix}(${POSTFIX})` as const;
export const decorateName = (mask: string, name: string) => mask.replace(POSTFIX, name);
export const undecorateName = (mask: string, decoratedName: string) => {
    const [start, end] = mask.split(POSTFIX);
    return decoratedName.replace(start, '').replace(end, '');
};
