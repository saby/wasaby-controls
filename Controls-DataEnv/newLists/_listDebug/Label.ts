/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import type { Slice } from 'Controls-DataEnv/slice';

const SYNTHETIC_NAME = '_dataSyntheticStoreId';

let LABELS_COUNT: number;
let LABELS: WeakMap<
    Slice,
    {
        name: string;
        isSynthetic: boolean;
    }
>;
(() => {
    LABELS_COUNT = 0;
    LABELS = new WeakMap();
})();

export function initLabel(instance: Slice, name: string): string {
    let resultName: string;

    if (name === SYNTHETIC_NAME) {
        LABELS_COUNT++;
        resultName = `SYNTHETIC #${LABELS_COUNT}`;
        LABELS.set(instance, {
            name: resultName,
            isSynthetic: true,
        });
    } else {
        resultName = name;
        LABELS.set(instance, { name, isSynthetic: false });
    }

    return resultName;
}

export function deleteLabel(instance: Slice): void {
    LABELS.delete(instance);
}

export function getLabel(instance: Slice): string {
    const info = LABELS.get(instance);
    if (info) {
        return info.name;
    }

    return '';
}
