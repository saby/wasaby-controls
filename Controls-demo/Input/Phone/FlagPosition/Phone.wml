<div class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center">
<div class="controlsDemo__cell" data-qa="Controls-demo_Input_Phone_FlagPosition__cell-start">
    <div class="controls-text-label">flagPosition = start</div>
    <Controls.input:Phone attr:class="ControlsDemo-PhoneInput_phoneField"
                          attr:data-qa="Controls-demo_Input_Phone_FlagPosition__start"
                          onlyMobile="{{true}}"
                          flagVisible="{{true}}"
                          fontSize="xl"
                          inlineHeight="2xl"
                          placeholder="+7 Телефон"
                          flagPosition="start"/>
</div>
<div class="controlsDemo__cell" data-qa="Controls-demo_Input_Phone_FlagPosition__cell-end">
    <div class="controls-text-label">flagPosition = end</div>
    <Controls.input:Phone attr:class="ControlsDemo-PhoneInput_phoneField"
                          attr:data-qa="Controls-demo_Input_Phone_FlagPosition__end"
                          onlyMobile="{{true}}"
                          flagVisible="{{true}}"
                          fontSize="xl"
                          inlineHeight="2xl"
                          placeholder="+7 Телефон"
                          flagPosition="end"/>
</div>
</div>
