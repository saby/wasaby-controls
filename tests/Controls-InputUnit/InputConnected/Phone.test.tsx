/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as DefaultValue } from 'Controls-Input-demo/InputConnected/Phone/DefaultValue';
import { default as FlagPosition } from 'Controls-Input-demo/InputConnected/Phone/FlagPosition';
import { default as FlagVisible } from 'Controls-Input-demo/InputConnected/Phone/FlagVisible';
import { default as Label } from 'Controls-Input-demo/InputConnected/Phone/Label';
import { default as OnlyMobile } from 'Controls-Input-demo/InputConnected/Phone/OnlyMobile';

describe('Controls-Input/InputConnected:Phone', () => {
    setTestID(148321624);

    let container = null;

    beforeEach(() => {
        jest.useFakeTimers('modern');
        jest.setSystemTime(new Date(1996, 5, 21));
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        jest.useRealTimers();
        container.remove();
        container = null;
    });

    it('defaultValue', async () => {
        render(getContent(DefaultValue), { container });
        expect(container).toMatchSnapshot();
    });
    it('FlagPosition', async () => {
        render(getContent(FlagPosition), { container });
        expect(container).toMatchSnapshot();
    });
    it('FlagVisible', async () => {
        render(getContent(FlagVisible), { container });
        expect(container).toMatchSnapshot();
    });
    it('label', async () => {
        render(getContent(Label), { container });
        expect(container).toMatchSnapshot();
    });
    it('OnlyMobile', async () => {
        render(getContent(OnlyMobile), { container });
        expect(container).toMatchSnapshot();
    });
});
