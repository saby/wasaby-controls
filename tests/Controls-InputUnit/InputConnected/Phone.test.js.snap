// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/InputConnected:Phone FlagPosition 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center"
  >
    <div
      class="controlsDemo__wrapper controlsDemo__flex"
    >
      <div
        class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
      >
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                flagPosition = start
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                name="$$wasaby$async$loading"
              />
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                flagPosition = end
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
              <div
                name="$$wasaby$async$loading"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Phone FlagVisible 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center"
  >
    <div
      class="controlsDemo__wrapper controlsDemo__flex"
    >
      <div
        class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
      >
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                flagVisible = true
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                name="$$wasaby$async$loading"
              />
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                flagVisible = false
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Phone OnlyMobile 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center"
  >
    <div
      class="controlsDemo__wrapper controlsDemo__flex"
    >
      <div
        class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
      >
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined "
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                onlyMobile = true
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                />
                <div
                  class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined "
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                onlyMobile = false
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                />
                <div
                  class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Phone defaultValue 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center"
  >
    <div
      class="controlsDemo__wrapper controlsDemo__flex"
    >
      <div
        class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
      >
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                defaultValue = 2
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="8 (800) 335-35-35"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-col undefined "
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                defaultValue is empty
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                />
                <div
                  class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Phone label 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo__flex ws-flex-column ws-align-items-center"
  >
    <div
      class="controlsDemo__wrapper controlsDemo__flex"
    >
      <div
        class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default controls-margin_top-m controls-margin_top-m controls-Input-connected_filled tw-w-full"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
            >
              <input
                autocapitalize="off"
                autocomplete="off"
                autocorrect="off"
                class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                inputmode="tel"
                name="ws-input_1996-06-20"
                spellcheck="true"
                tabindex="0"
                type="text"
                value="+7 (999) 999-99-99"
              />
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Введите номер
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                label top
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default controls-margin_top-m"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
        >
          <a
            class="controls-Label controls-Label_underline_none tw-min-w-0 tw-flex-shrink"
          >
            <div
              class="controls-Label__wrapper"
            >
              <div
                class="controls-Label__caption controls-fontsize-m  controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
              >
                label start
              </div>
              
            </div>
          </a>
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default controls-input_label_min-input-content controls-margin_top-m"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
        >
          <span
            class="controls-icon_size-s controls-icon_style-secondary icon-SabyBird controls-icon controls-margin_right-s"
          />
          <div
            class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-m controls-Render-inlineheight-m  controls-Phone controls_Phone_theme-default controls-margin_top-m"
            tabindex="0"
          >
            <div
              class="controls-Render__wrapper"
            >
              <span
                class="controls-Render__baseline"
              >
                ﻿
              </span>
              <div
                class="controls-InputBase__field controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
                data-qa="controls-Render__field"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                  inputmode="tel"
                  name="ws-input_1996-06-20"
                  spellcheck="true"
                  tabindex="0"
                  type="text"
                  value="+7 (999) 999-99-99"
                />
                <div
                  class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
                >
                  <div
                    class="controls-Render__placeholder controls-Render__placeholder_overflow"
                  >
                    Введите номер
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
